File: src/backend/app/models/strategy.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Enum
from sqlalchemy.sql import func
from app.db.database import Base
import enum


class StrategyStatus(str, enum.Enum):
    DRAFT = "draft"
    ACTIVE = "active"
    PAUSED = "paused"
    ARCHIVED = "archived"


class Strategy(Base):
    __tablename__ = "strategies"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False)
    description = Column(Text)
    code = Column(Text, nullable=False)
    status = Column(Enum(StrategyStatus), default=StrategyStatus.DRAFT)
    user_id = Column(Integer, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

File: src/frontend/src/types/strategy.ts
// 策略类型定义

export type StrategyStatus = 'draft' | 'active' | 'inactive' | 'archived'

export interface Strategy {
  id: number
  name: string
  description?: string
  status: StrategyStatus
  symbol: string
  parameters: Record<string, any>
  createdAt: string
  updatedAt: string
}

export interface CreateStrategyDto {
  name: string
  description?: string
  symbol: string
  parameters: Record<string, any>
}

export interface UpdateStrategyDto {
  name?: string
  description?: string
  status?: StrategyStatus
  parameters?: Record<string, any>
}
